src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerCache.cs

```diff
@@ -56,22 +56,28 @@ public Entry GetCacheEntry(ControllerContext controllerContext)
                 return entry;
```

Comment:
Which method is called by the invoker? For instance calling `GetFilters` can still double-execute the providers. I think we've got more cleanup to do here.